remediation_worker/jobs/kinesis_encrypt_stream/kinesis_encrypt_stream.py

```diff
@@ -0,0 +1,173 @@
+            (dict) Dictionary of parsed parameters.
+        """
+        remediation_entry = json.loads(payload)
+        notification_info = remediation_entry.get("notificationInfo", None)
+        finding_info = notification_info.get("FindingInfo", None)
+
+        stream_name = finding_info.get("ObjectId", None)
+        region = finding_info.get("Region", None)
+
+        if stream_name is None:
+            logging.error("Missing parameters for 'STREAM_NAME'.")
+            raise Exception("Missing parameters for 'stream_name'.")
+
+        if region is None:
+            logging.error("Missing parameters for 'REGION'.")
+            raise Exception("Missing parameters for 'REGION'.")
+
+        logging.info("parsed params:")
+        logging.info("  stream_name: %s", stream_name)
+        logging.info("  region: %s", region)
+
+        return {
+            "stream_name": stream_name,
+            "region": region
+        }
+
+    def remediate(self, stream_name, client, region):
+        """Encrypts unencrypted kinesis data streams.
+        Args:
+            stream_name ([str]): The name of the kinesis stream.
+            client: Instance of the AWS boto3 client.
+        """
+
+        logging.info(
+            "Beginning kinesis data stream encryption evaluation."
+        )
+
+        # Check to see if kinesis data stream is encrypted
+        encryption_type = self.check_encryption(client, stream_name)
+
+        # If kinesis data stream is not encrypted
+        if encryption_type == "NONE":
+            self.enable_encryption(client, stream_name)
+        # If encryption is enabled
+        # We have nothing to do.
+        # log and exit
+        elif encryption_type == "KMS":
+            logging.info(
+                "Kinesis data stream %s is already encrypted, taking no action.",
+                stream_name
+            )
+            sys.exit(0)
+        # If anything else returns,
+        # log and exit
+        else:
+            logging.error(
+                "An unknown encryption type returned for kinesis data stream %s. Exiting.",
+                stream_name
+            )
+            sys.exit(1)
+
+    def check_encryption(self, client, stream_name):
+        """Checks the kinesis data stream encryption setting.
+        Args:
+            client: Instance of the AWS boto3 client
+            stream_name ([str]): The name of the kinesis stream.
+        Returns:
+            [string]: "NONE" if no encryption is enabled.
+        """
+
+        # Check if stream is encrypted
+        try:
+            encryption_type = client.describe_stream(
+                StreamName=stream_name)["StreamDescription"]["EncryptionType"]
+        except client.exceptions.ResourceNotFoundException as found_error:
+            logging.error(
+                "A failure occured with error: %s. Check if the resource exists.",
+                format(found_error.response['Error']['Code'])
+            )
+            sys.exit(1)
+
```

Comment:
Can you add another except block and log the error returned by AWS other than ResourceNotFoundException.

Reply:
can't  help but notice that format() is meaningless here